pandas/core/generic.py

```diff
@@ -3096,7 +3096,8 @@ def to_csv(
             compression mode is 'infer' and `path_or_buf` is path-like, then
             detect compression mode from the following extensions: '.gz',
             '.bz2', '.zip' or '.xz'. (otherwise no compression). If dict given
-            and mode is 'zip' or inferred as 'zip', other entries passed as
+            and mode is one of {'zip', 'gzip', 'bz2'}, or inferred as
```

Comment:
IIRC there is an example in io.rst that does this, can you check and augment that example (and if not pls create one.

Reply:
Good catch -- found it and updated.